### PR TITLE
Exact match behavior change

### PR DESCRIPTION
**Pull request by density215**
A query with match_type `ExactMatch` returns no less or more specifics if there is no exactly matching prefix. We are changing this behavior to include the less and/or more specifics (if the user specified it in the `MatchOptions`) even if the query itself returns no matching prefix (`EmptyMatch`).

This solves https://github.com/NLnetLabs/rotonda-runtime/issues/79

### CODE REVIEW

File: src/rotonda_store.rs [hunk 1]
```diff
@@ -65,10 +65,23 @@ impl<'a> std::fmt::Debug for Strides<'a> {
...
+/// 
+/// See [MultiThreadedStore::match_prefix] for more details.
 pub struct MatchOptions {
+    /// The requested [MatchType]
     pub match_type: MatchType,
+    /// unused
     pub include_all_records: bool,
+    /// whether to include all less-specific records in the query result
     pub include_less_specifics: bool,
+    // whether to include all more-specific records in the query result
```
Comment (ximon18) [inline src/rotonda_store.rs:84]: Missing forward slash in the comment.

File: src/rotonda_store.rs [hunk 2]
```diff
@@ -274,12 +287,26 @@ impl<'a, M: routecore::record::Meta> std::fmt::Display
 
 //------------- QueryResult -------------------------------------------------
 
+/// The type that is returned by a query.
+/// 
+/// This is the result type of a query. It contains the prefix record that was
+/// found in the store, as well as less- or more-specifics as requested.
+/// 
+/// See [MultiThreadedStore::match_prefix] for more details.
+
```
Comment (ximon18) [inline src/rotonda_store.rs:296]: Remove these blank lines?

File: src/rotonda_store.rs [hunk 3]
```diff
@@ -65,10 +65,23 @@ impl<'a> std::fmt::Debug for Strides<'a> {
...
+/// `match_prefix` method on the store.
+/// 
+/// Note that the `match_type` field may be different from the actual
+/// `MatchType` returned from the result. 
+/// 
+/// See [MultiThreadedStore::match_prefix] for more details.
 pub struct MatchOptions {
+    /// The requested [MatchType]
     pub match_type: MatchType,
+    /// unused
```
Comment (ximon18) [inline src/rotonda_store.rs:80]: s/unused/Unused/

File: src/rotonda_store.rs [hunk 4]
```diff
@@ -65,10 +65,23 @@ impl<'a> std::fmt::Debug for Strides<'a> {
...
     pub include_all_records: bool,
+    /// whether to include all less-specific records in the query result
```
Comment (ximon18) [inline src/rotonda_store.rs:82]: s/whether/Whether/

Comment (ximon18) [inline src/rotonda_store.rs:82]: Same comment about "all" as below.

**Comment by ximon18** [comment]
I tested this by incorporating this PR into a local copy of the `rotonda-store` branch that `rotonda-runtime` depends on. See https://github.com/NLnetLabs/rotonda-runtime/pull/81.

File: src/rotonda_store.rs (hunk 1 above)
Comment (density215) [reply src/rotonda_store.rs:84]: There's going to be an option where you the user can specify to receive only more-specifics within a range.

**Pull request merged**